app/controllers/teachers_controller.rb

```diff
@@ -100,7 +100,10 @@ def update
       @teacher.school = @school
     else
       @school.update(school_params) if school_params
-      @school.save!
+      unless @school.save
+        flash[:alert] = "An error occurred: #{@school.errors.full_messages.join(', ')}"
+        render "new" && return
```

Comment:
```suggestion
        render "edit" && return
```

Or am i missing something?